### DISCUSSION TITLE
I have been trying to get my listeners to work. Please help.

### DISCUSSION THREAD

**Discussion by VaibhavTheVicar** [awspring/spring-cloud-aws#1056] [Q&A]
Read this in sqs listener docs.

/**
 * Methods with this annotation will be wrapped by a {@link io.awspring.cloud.sqs.listener.MessageListener} or
 * {@link io.awspring.cloud.sqs.listener.AsyncMessageListener} and set to a
 * {@link io.awspring.cloud.sqs.listener.MessageListenerContainer}.
 * <p>
 * Each method will be handled by a different container instance, created by the specified {@link #factory()} property.
 * If not specified, a default factory will be looked up in the context.
 * <p>
 * When used in conjunction with Spring Boot and auto configuration, the framework supplies a default
 * {@link io.awspring.cloud.sqs.config.SqsMessageListenerContainerFactory} and a
 
 
 When I fire a message to queue "test-queue-demo", I see logs from the custom customSqsMessageListenerContainer() bean, I created for testing.. but not from the @SqsListener, method. Ideally.. a MessageListenerContainer instance should have been ceared and the code in the listener annotated method must trigger, but it does not.. as in above doc.. but looks like it doesn't.. Have been on this for hours.. please assist.
 
 @tomazfernandes @alainsahli @MatejNedic please some insight.
 
 //    }
 ```
   @Bean("customSqsListenerContainerFactory")
    public SqsMessageListenerContainerFactory<Object> customSqsListenerContainerFactory(
            SqsAsyncClient sqsAsyncClient
    ) {
        // Your custom configuration
        return SqsMessageListenerContainerFactory.builder()
                .configure(options -> options
                        .messageVisibility(Duration.ofSeconds(100))
                        .maxMessagesPerPoll(10)
                        .pollTimeout(Duration.ofSeconds(10))
                        .queueNotFoundStrategy(QueueNotFoundStrategy.FAIL)
                )
                .sqsAsyncClient(sqsAsyncClient)
                .messageListener(e -> {
                    System.out.println("Caught in factory " + e);
                })
                .build();
    }

    @Bean
    MessageListenerContainer customSqsMessageListenerContainer(
            @Qualifier("customSqsListenerContainerFactory") SqsMessageListenerContainerFactory sqsMessageListenerContainerFactory
    ) {
        return sqsMessageListenerContainerFactory.createContainer("test-queue-demo");
    }

    @SqsListener(value = "test-queue-demo", factory = "customSqsListenerContainerFactory")
    public void customSqsListenerMethod(String message) {
        // Your method implementation
        System.out.println("Listened in SqsListener " + message);
    }
```

**Comment by tomazfernandes** [comment]
@VaibhavTheVicar, I'm not sure I understand this.

Why are you creating two listeners listening to the same queue? If the other listener catches the message first, the SqsListener won't have a message to listen to.

Please leave only the `@SqsListener` one and let me know if that works. Thanks.
